web_tree_dynamic_colored_field/README.rst

```diff
@@ -0,0 +1,133 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.png
+   :target: https://www.gnu.org/licenses/agpl
+   :alt: License: AGPL-3
+
+============================
+Colorize field in tree views
+============================
+
+This module aims to add support for dynamically coloring fields in tree view
+according to data in the record.
+
+It provides attributes on fields with the similar syntax as the ``colors`` attribute
+in tree tags.
+
+Further, it provides a ``color_field`` attribute on tree tags's ``colors`` to use
+a field's value as color.
+
+Features
+========
+
+* Add attribute ``bg_color`` on field's ``options`` to color background of a cell in tree view
+* Add attribute ``fg_color`` on field's ``options`` to change text color of a cell in tree view
+* Add attribute ``color_field`` on the tree element's ``colors`` to use as color
+
+Usage
+=====
+
+* In the tree view declaration, put ``options='{"bg_color": "red: customer==True"}`` attribute in the ``field`` tag::
+
+    ...
+    <field name="arch" type="xml">
+        <tree string="View name">
+            ...
+            <field name="name" options='{"bg_color": "red: customer == True"}'/>
+            ...
+        </tree>
+    </field>
+    ...
+
+    With this example, column which renders 'name' field will have its background colored in red.
+
+* In the tree view declaration, put ``options='{"fg_color": "white:customer == True"}'`` attribute in the ``field`` tag::
+
+    ...
+    <field name="arch" type="xml">
+        <tree string="View name">
+            ...
+            <field name="name" options='{"fg_color": "white:customer == True"}'/>
+            ...
+        </tree>
+    </field>
+    ...
+
+    With this example, column which renders 'name' field will have its text colored in white on a customer records.
+
+* In the tree view declaration, use ``options='"color_field": "my_color"'`` attribute in the ``tree`` tag::
+
+    ...
+    <field name="arch" type="xml">
+        <tree string="View name" colors="color_field: my_color" >
+            ...
+            <field name="my_color" invisible="1"/>
```

Comment:
Can we use HEX color codes in this field or it should always be the color name?
Correct me if I understood it in wrong way.

Reply:
You can see in the demo data that the HEX colors also can be used.